craft_application/manifest.py

```diff
@@ -0,0 +1,96 @@
+# This file is part of craft-application.
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Functions to generate the craft manifest of an artifact."""
+
+import hashlib
+import pathlib
+from datetime import datetime, timezone
+from typing import Any
+
+from craft_application.models import BuildInfo, Project
+from craft_application.models.manifest import (
+    CraftManifest,
+    Hashes,
+    ProjectManifest,
+    SessionArtifactManifest,
+)
+
+
+def from_packed_artifact(
```

Comment:
This seems like it would make more sense as a classmethod on `ProjectManifest`.

Reply:
I thought about it; I guess I wanted to separate model and implementation a bit more. I can move all this code to the models though no problem
what do you think @cmatsuoka ?

Reply:
Classmethod seems more appropriate in this case (we have other instances of "from" constructors in our projects).

Reply:
Updated - moved all functions to staticmethods on the models